misc/testerina/modules/testerina-runtime/src/main/java/org/ballerinalang/test/runtime/BTestRunner.java

```diff
@@ -603,32 +643,42 @@ private void stopSuite(TestSuite suite, Scheduler scheduler, Class<?> initClazz,
-                                      Scheduler scheduler) throws ClassNotFoundException {
-        Class<?> functionClass = classLoader.loadClass(suite.getTestUtilityFunctions().get(functionName));
-        TesterinaFunction testerinaFunction = new TesterinaFunction(functionClass, functionName, scheduler);
-        return testerinaFunction.invoke();
+                                      Scheduler scheduler) {
+        try {
+            Class<?> functionClass = classLoader.loadClass(suite.getTestUtilityFunctions().get(functionName));
+            TesterinaFunction testerinaFunction = new TesterinaFunction(functionClass, functionName, scheduler);
+            return testerinaFunction.invoke();
+        } catch (ClassNotFoundException e) {
+            return e;
+        }
     }
 
-    public void invokeTestFunction(TestSuite suite, String functionName, ClassLoader classLoader,
-                                   Scheduler scheduler, Class<?>[] types, Object[] args) throws ClassNotFoundException {
-        Class<?> functionClass = classLoader.loadClass(suite.getTestUtilityFunctions().get(functionName));
-        TesterinaFunction testerinaFunction = new TesterinaFunction(functionClass, functionName, scheduler);
-        testerinaFunction.invoke(types, args);
+    public Object invokeTestFunction(TestSuite suite, String functionName, ClassLoader classLoader,
+                                     Scheduler scheduler, Class<?>[] types, Object[] args) {
+        try {
+            Class<?> functionClass = classLoader.loadClass(suite.getTestUtilityFunctions().get(functionName));
+            TesterinaFunction testerinaFunction = new TesterinaFunction(functionClass, functionName, scheduler);
+            return testerinaFunction.invoke(types, args);
+        } catch (ClassNotFoundException e) {
+            return e;
+        }
+
     }
 
     private String formatErrorMessage(Throwable e) {
         String message;
-        if (e.getCause() instanceof BError) {
-            try {
-                message = ((BError) e.getCause()).getPrintableStackTrace();
-            } catch (ClassCastException castException) {
-                // throw the exception to top
+        try {
+            if (e instanceof BError) {
+                message = ((BError) e).getPrintableStackTrace();
+            } else if (e instanceof Exception | e instanceof Error) {
+                RuntimeUtils.silentlyLogBadSad(e);
+                message = TesterinaUtils.getPrintableStackTrace(e);
+            } else {
                 throw new BallerinaTestException(e);
             }
-        } else if (e instanceof BallerinaTestException) {
-            throw (BallerinaTestException) e;
-        } else {
-            throw new BallerinaTestException(e);
+        } catch (ClassCastException classCastException) {
+            // If an unhandled error type is passed to format error message
+            throw new BallerinaTestException(classCastException);
```

Comment:
Can't we simplify this method using direct returns? (As the `message` variable is used only within the `try` block)

Reply:
Refactored to use only a direct return for all instances where message was assigned